esmvalcore/preprocessor/_multimodel.py

```diff
@@ -68,6 +95,9 @@ def _compute_statistic(data, statistic_name):
         statistic_function = np.ma.max
     elif statistic_name == 'min':
         statistic_function = np.ma.min
+    elif statistic_name.startswith('p'):
```

Comment:
Hmm.. I see the point, but this solution doesn't work. If you except it and pass, it will never reach the final `else`, and you will get `UnboundLocalError: local variable 'statistic_function' referenced before assignment`. So we should raise rather than pass. You could catch the `ValueError` and raise `NotImplementedError`, but then why not just keep the `ValueError`? That error message also provides a better pointer as to what's wrong. I'd suggest to just leave it for now, and sort it out once we get this statistic that starts with p.

Reply:
Maybe you could do something like this to make it more selective:
```suggestion
    elif re.match("^p\d+[\.]?\d*$", statistic):
```

Reply:
It would also be good to add a check that the values are valid in esmvalcore/_recipe_checks.py or in esmvalcore/recipe_schema.yml, so users don't spend ages running a recipe to regrid a 100 models only to find that they typed the wrong name for the statistic they would like to compute.

Reply:
Okay, what about the following: `^(p\d{1,2})(\.\d*)?$`: starts with `p` followed by 1 or 2 digits, which may or may not be followed by a decimal part consisting of a period followed by 0 to &infin;  digits. Works for most cases like `p1`, `p1.`, `p1.1`, `p1.11`, `p11`, ..., and also for `p0`, but not for `p.1`, `p.11`, or `p100`.

Reply:
You could also just accept anything starting with `p` and then try converting whatever you get to a valid percentile and raise a nice error message if that fails, it's the last option to try anyway.